index.bs

```diff
@@ -1966,6 +1966,40 @@ See also:
 
 * <a href="https://www.w3.org/2001/tag/doc/promises-guide">Writing Promise-Using Specifications</a>
 
+<h4 id="sync-callbacks">When to use callback functions instead</h4>
+
+<p>If an algorithm requires that callback functions respond
+synchronously (e.g. to take or veto some action), then that is a
+synchronous API, and should not be modelled with a promise. Instead,
+use an explicit callback or event.</p>
+
+<p>In [Promise reaction callbacks](https://tc39.es/ecma262/multipage/control-abstraction-objects.html#sec-promisereaction-records),
+you may not require action within
+a fixed number of microtasks.</p>
+
+<p>This protects promise composition: patterns like <code>await</code>,
+wrapping helper functions, logging, <code>Promise.race()</code>, or multiple
+<code>.then()</code> branches all queue an extra microtask and must
+not significantly change the behavior of the code they wrap.</p>
+
+<div class=example>
+If developer-supplied code needs to be able to call <code>event.{{Event/preventDefault()}}</code>
+or choose a response object with
+<code>fetchEvent.{{FetchEvent/respondWith()}}</code>,
+it needs to be a callback rather than a reaction passed to <code>Promise.{{Promise/then()}}</code>.
+</div>
+<div class="example">
+A non-event example is the <code>captureController.{{CaptureController/setFocusBehavior()}}</code>
+method to push a window the user has chosen to screen-capture, to
+the front. Applications can decide this based on the window chosen.
+But any delay risks click-jacking.
```

Comment:
```suggestion
The challenge is that allowing delay creates an opportunity for click-jacking attacks from sites.
```

A link defining click-jacking would be helpful.

Reply:
https://html.spec.whatwg.org/#writing-secure-applications-with-html is a section reference with a definition, but the spec doesn't have an anchor on the definition itself.